docker-images/operator/scripts/dynamic_resources.sh

```diff
@@ -15,3 +15,26 @@ function get_heap_size {
     echo "${CONTAINER_HEAP_MAX}"
   fi
 }
+
+function set_xmx_xms {
+  MAX_HEAP=`get_heap_size`
```

Comment:
MAX_HEAP was _supposed_ to be a heuristic for what to set Xmx to if it was not explicitly configured. Recent versions of `java` have their own heuristic for figuring out the heap size to use inside a container, based on the container memory limit. IIRC the `get_heap_size` dates from before that was in java 8, but there was another motivation of `get_heap_size`: In the Kafka container we don't really want the JVM ergonomics to be based on the container memory limit, because we want plenty of 'spare' memory to be used by the page cache. So we needed a way to be able to configure the memory in a container-aware way, but potentially with fine control over how much of the container memory to use. At the time the only way to do this was using `-XX:MaxRAMFraction` which suffered from the issue described in https://bugs.openjdk.java.net/browse/JDK-8186315. As you can see that was fixed in Java 10. So once we move to Java 11 we should be able to simplify this.

So today, with Java memory ergonomics being container aware, I think the Kafka part is the only thing we still need to worry about. I.e. we probably don't need to call `get_heap_size` at all for non-Kafka containers. 

Now, if you read `get_heap_size` it is expecting two arguments, which are not being passed here (though they are being passed in `kafka_run.sh`, where the DYNAMIC_HEAP_FRACTION is our workaround for the fact that Java 8 lacks the `-XX:MaxRAMPercentage` option mentioned in the linked issue).

So I think we should remove this `MAX_HEAP` stuff from `launch_java.sh` entirely. If we need to do any meddling with Xmx etc it should be done in the script which calls `launch_java.sh`.